### PR TITLE
Add workflow dispatch to upload condition

### LINKED ISSUE #59: The workflow only allows upload to pypi on push events
not for instance `workflow_dispatch`

### PR DESCRIPTION
**Pull request by Cadair**
Fixes #59

### CODE REVIEW

**Comment by Cadair** [comment]
Happy to add other things here, release definitely makes sense.

**Comment by ConorMacBride** [comment]
I had a look through the list, and these two are probably enough:
- [`create`](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#create) — A Git branch or tag is created.
- [`release`](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#release) — Activity related to a release.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -102,7 +102,7 @@ jobs:
           fi
         env:
           UPLOAD_TO_PYPI: ${{ inputs.upload_to_pypi }}
-          UPLOAD_TAG: ${{ startsWith(inputs.upload_to_pypi, 'refs/tags/') && github.event_name == 'push' && startsWith(github.event.ref, inputs.upload_to_pypi) }}
+          UPLOAD_TAG: ${{ startsWith(inputs.upload_to_pypi, 'refs/tags/') && (github.event_name == 'push' || github.event_name == 'workflow_dispatch') && startsWith(github.ref, inputs.upload_to_pypi) }}
```
Comment (ConorMacBride) [inline .github/workflows/publish.yml:105]:
```suggestion
          UPLOAD_TAG: ${{ startsWith(inputs.upload_to_pypi, 'refs/tags/') && (github.event_name == 'push' || github.event_name == 'workflow_dispatch' || github.event_name == 'release' || github.event_name == 'create') && startsWith(github.ref, inputs.upload_to_pypi) }}
```

File: .github/workflows/publish_pure_python.yml [hunk 2]
```diff
@@ -86,7 +86,7 @@ jobs:
           fi
         env:
           UPLOAD_TO_PYPI: ${{ inputs.upload_to_pypi }}
-          UPLOAD_TAG: ${{ startsWith(inputs.upload_to_pypi, 'refs/tags/') && github.event_name == 'push' && startsWith(github.event.ref, inputs.upload_to_pypi) }}
+          UPLOAD_TAG: ${{ startsWith(inputs.upload_to_pypi, 'refs/tags/') && (github.event_name == 'push' || github.even_name == 'workflow_dispatch') && startsWith(github.ref, inputs.upload_to_pypi) }}
```
Comment (ConorMacBride) [inline .github/workflows/publish_pure_python.yml:89]:
```suggestion
          UPLOAD_TAG: ${{ startsWith(inputs.upload_to_pypi, 'refs/tags/') && (github.event_name == 'push' || github.event_name == 'workflow_dispatch' || github.event_name == 'release' || github.event_name == 'create') && startsWith(github.ref, inputs.upload_to_pypi) }}
```

**Review by ConorMacBride — APPROVED** [review]
This looks good. I'll merge, create a release (pytest-mpl is pinned to v1) and dispatch the pytest-mpl workflow on its latest tag to get the release on pypi.

**Pull request merged**